### PR TITLE
format support

### PR DESCRIPTION
**Pull request by Yvonne-Aizawa**
with ollama 0.1.9  you can send if you want the output to be formatted.
for now it only supports JSON. but when left empty it generates as normal.

I added a parameter to GenerationRequest to specify if it needs to be formatted as json. when it is None it will leave out the parameter and generate as normal. I made it an enum as i seems like the format parameter might get more types and it would be easy to add new ones to it.

### CODE REVIEW

**Comment by pepperoni21** [comment]
Hey, I don't think we need to support other formats than json since we're using Serde, the user does not matter in what format we get the data as they would just access it through the Rust bindings that this library provides.

**Comment by Yvonne-Aizawa** [comment]
I do not know if i understand it right. But this is how ollama works now [docs](https://github.com/jmorganca/ollama/blob/main/docs/api.md#json-mode). it forces the llm to return valid json. this just got added in ollama 0.1.9. you still recieve the data as a string. but the string would be a valid json object

**Comment by Yvonne-Aizawa** [comment]
example with formatting to json on
> what is the weather like?
{"message": "I'm just an AI, I don't have access to real-time weather information. However, I can tell you that the weather can vary depending on your location and time of year. Could you please provide me with more context or specify a location you are interested in?", "request_id": 123456}

example with formatting none
> what is the weather like?
I'm just an AI, I don't have real-time access to current weather conditions. However, I can tell you the typical weather patterns for a place if you provide me with the location. Please let me know the city or region you are interested in, and I will do my best to give you an idea of what the weather is like there.

File: README.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ if let Ok(res) = res {
 let model = "llama2:latest".to_string();
 let prompt = "Why is the sky blue?".to_string();
 
-let mut stream = ollama.generate_stream(GenerationRequest::new(model, prompt)).await.unwrap();
+let mut stream = ollama.generate_stream(GenerationRequest::new(model, prompt,None)).await.unwrap();
```
Comment (pepperoni21) [inline README.md:40]: I think the format should be None by default without having to add it to the new method, like the other optional fields, as it meets a very particular need.

File: src/generation/completion/request.rs [hunk 2]
```diff
@@ -13,18 +13,25 @@ pub struct GenerationRequest {
...
-    pub fn new(model_name: String, prompt: String) -> Self {
+    pub fn new(model_name: String, prompt: String, format: Option<FormatEnum>) -> Self {
         Self {
             model_name,
             prompt,
             options: None,
             system: None,
             template: None,
             context: None,
+            //format value for now just json
```
Comment (pepperoni21) [inline src/generation/completion/request.rs:33]:
```suggestion
            // The format to return a response in. Currently the only accepted value is `json`
```

File: tests/generation.rs [hunk 3]
```diff
@@ -42,7 +43,22 @@ async fn test_generation() {
...
+    let _ = ollama
+        .generate(GenerationRequest::new(
+            "llama2:latest".to_string(),
+            PROMPT.into(),
+            Some(FormatEnum::Json)
+            
+        ))
+        .await
+        .unwrap();
+}
```
Comment (pepperoni21) [inline tests/generation.rs:64]: I don't think we need a test for that

**Review by pepperoni21 — CHANGES_REQUESTED** [review]

**Comment by Yvonne-Aizawa** [comment]
removed parameter from the new() method. and updated code to reflect it
altered the comment to explain better what the field does
removed the test from generation.rs

**Pull request merged**